x-pack/plugin/snapshot-repo-test-kit/qa/s3/src/javaRestTest/java/org/elasticsearch/repositories/blobstore/testkit/S3SnapshotRepoTestKitIT.java

```diff
@@ -35,7 +35,6 @@ public class S3SnapshotRepoTestKitIT extends AbstractSnapshotRepoTestKitRestTest
         .setting("logger.org.elasticsearch.repositories.s3", "TRACE")
         .setting("logger.org.elasticsearch.repositories.blobstore.testkit", "TRACE")
         .setting("logger.com.amazonaws.request", "DEBUG")
-        .setting("logger.org.apache.http.wire", "DEBUG")
```

Comment:
@DaveCTurner With respect to #102294, is DEBUG required for `org.apache.http.wire` or would it be fine to remove this? We are trying to prevent this to not risk leaking creds or similar.

Reply:
Yep sorry we definitely need to preserve the ability to trace at this level in tests and in certain production situations where there's some debate about exactly how the customer's allegedly-S3-compatible repository is behaving differently from S3. Could we gate this behind `org.elasticsearch.transport.NetworkTraceFlag#TRACE_ENABLED` like we do with the other low-level tracing options?

Reply:
Thanks @DaveCTurner, that makes perfectly sense to me. I'll change as suggested (including documentation) and will also restrict the AWS request logger. 
@rjernst Are you ok to move forward as suggested?

Reply:
@rjernst Kind ping, are you ok to gate the behavior added here by `NetworkTraceFlag#TRACE_ENABLED`?

Reply:
+1 to guarding it behind the network trace flag.